frappe/core/page/calendar/calendar.js

```diff
@@ -0,0 +1,290 @@
+
+			// for mapping events into calendar
+			events: function(start, end, timezone, callback) {
+				start_param = get_system_datetime(start);
+				end_param = get_system_datetime(end)
+
+				var docinfo = '';
+				$('.module-sidebar-nav input:checked').each(function() {
+					docinfo += $(this).attr('value')+',';
+				});
+				//methd for fetching the events.
+				return frappe.call({
+					method: "frappe.core.page.calendar.calendar.get_master_calendar_events",
+					type: "GET",
+					args : {
+						'doctypeinfo': docinfo,
+						'start' : start_param,
+						'end': end_param
+					}
+				}).then(r => {
+					var events = [];
+					for (event in r["message"]){
+						//after fetching pushing and maping events on calendar
+						if($("input[value='"+r["message"][event].doctype+"']").prop("checked")){
+							events.push({
+								start: r["message"][event].start,
+								end: r["message"][event].end,
+								id: r["message"][event].id,
+								title: r["message"][event].title,
+								doctype: r["message"][event].doctype,
+								color: r["message"][event].color, 
+								textColor: r["message"][event].textColor,
+								description: r["message"][event].description
+							});
+						}
+					}
+					callback(events)
+				});
+			},
+
+			//Drag event (to create new Event)modal fade in
+			select: function(startDate, endDate, jsEvent, view) {
+				var checkboxes = $('.module-sidebar-nav input:checked');
+				var interval = endDate-startDate;
+
+				//identifying single day event
+				if(interval > 86400000){
+					if (checkboxes.length == 1){
+						frappe.new_doc(checkboxes[0].value);
+					}
+					else if (checkboxes.length > 1){
+						options = '';
+						$('.module-sidebar-nav input:checked').each(function() {
+							options += '\n'+$(this).attr('value');
+						});
+						
+						//select doctype to create event
+						frappe.prompt([
+							{'fieldname': 'Doctype', 'fieldtype': 'Select', 'options': options,'label': 'Doctype', 'reqd': 1}  
+						],
+						function(values){
+							frappe.new_doc(values.Doctype);
+							
+						},
+						'Select Doctype',
+						'Submit'
+						)
+			
+					}
+					else{
+						frappe.msgprint("Select doctype to create calendar event")
+					}
+				}
+			},
+
+			//Event click action 
+			eventClick: function(event,jsEvent) {
+				//removing popover if present
+				$(".popover.fade.bottom.in").remove();
+
+				var t = $(jsEvent.target)
+				// creating time html
+				if(event.allDay){
+					var timeHtml = "All Day"
+				}
+				else if(event.start.isSame(event.end, 'date', 'month', 'year')) {
+					var timeHtml = event.start.format('LT')+" to "+event.end.format('LT')
+				}
+				else if(event.start.isSame(event.end, 'month', 'year')){
+					var timeHtml = event.start.format("MMMM, ") + event.start.format('D')+" to "+ event.end.format('D')
+				}
+				else {
+					var timeHtml = event.start.format('Do MMMM')+" to "+ event.end.format('Do MMMM')
+				}
+
+				timing = "<div class='mt-5'>" +
```

Comment:
break these into small functions. In general aim for small functions for better readability

https://github.com/frappe/erpnext/wiki/Contribution-Guidelines

Reply:
done